napari/_qt/_qapp_model/injection/_qprocessors.py

```diff
@@ -28,8 +37,168 @@ def _add_plugin_dock_widget(
     viewer.window.add_dock_widget(widget, name=full_name)
 
 
+def _add_layer_data_tuples_to_viewer(
+    data: Union[tuple, list[tuple]],
+    return_type: Optional[Any] = None,
+    viewer: Optional[viewer.Viewer] = None,
+    source: Optional[dict] = None,
+) -> None:
+    from napari.utils.misc import ensure_list_of_layer_data_tuple
+
+    if viewer is None:
+        viewer = _provide_viewer()
+    if viewer and data is not None:
+        data = data if isinstance(data, list) else [data]
+        for datum in ensure_list_of_layer_data_tuple(data):
+            # then try to update a viewer layer with the same name.
+            if len(datum) > 1 and (name := datum[1].get('name')):
+                with suppress(KeyError):
+                    layer = viewer.layers[name]
+                    layer.data = datum[0]
+                    for k, v in datum[1].items():
+                        setattr(layer, k, v)
```

Comment:
Yes. I notice when I move things over to qt I get mypy errors, so I assumed mypy was not checking in the old location. I guess it could be the same for code coverage?

I don't think the processors are well. There is just one `test_processors.py`, which just checks for for error in `_add_layer_data_to_viewer`

Reply:
ahh yeah I remember we had an issue about adding more tests for these. Ok cool, I'm approving!

Reply:
ref:  #6847